collect_app/src/test/java/org/odk/collect/android/activities/FormMapActivityTest.java

```diff
@@ -0,0 +1,198 @@
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.android.controller.ActivityController;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.isA;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.odk.collect.android.activities.FormMapViewModelTest.testInstances;
+import static org.robolectric.Shadows.shadowOf;
+
+@RunWith(RobolectricTestRunner.class)
+public class FormMapActivityTest {
+    private ActivityController activityController;
+    private FormMapActivity activity;
+
+    private List<MapPoint> expectedPoints = Arrays.asList(new MapPoint(10, 125.6),
+            new MapPoint(10.1, 125.6), new MapPoint(10.1, 126.6),
+            new MapPoint(10.3, 125.6), new MapPoint(10.3, 125.7),
+            new MapPoint(10.4, 125.6));
+    private MapPoint currentLocation = new MapPoint(5, 5);
+
+    @Before public void setUpActivity() {
+        activityController = RobolectricHelpers.buildThemedActivity(FormMapActivity.class);
+        activity = (FormMapActivity) activityController.get();
+
+        TestInstancesRepository testInstancesRepository = new TestInstancesRepository(Arrays.asList(testInstances));
+        FormMapViewModel viewModel = new FormMapViewModel(FormMapViewModelTest.testForm1, testInstancesRepository);
+        activity.viewModelFactory = new TestFactory(viewModel);
+
+        activity.map = new TestMapFragment();
+
+        activityController.setup().visible().start().resume();
+    }
+
+    @Test public void startingFormMap_zoomsToFitAllInstanceMarkers_ifThereAreInstanceMarkers() {
+        TestMapFragment map = (TestMapFragment) activity.map;
+        assertThat(map.getZoomCount(), is(1));
+        assertThat(map.getLatestZoomPoint(), is(nullValue()));
+        assertThat(map.getLatestZoomBoundingBox(), is(expectedPoints));
+        assertThat(map.getLatestScaleFactor(), is(0.8));
+        assertThat(map.wasLatestZoomCallAnimated(), is(false));
+    }
+
+    @Test public void startingFormMap_doesNotZoom_ifThereAreNoInstanceMarkers_andLocationIsUnavailable() {
+        ActivityController controller = RobolectricHelpers.buildThemedActivity(FormMapActivity.class);
+        FormMapActivity activity = (FormMapActivity) controller.get();
+
+        TestInstancesRepository testInstancesRepository = new TestInstancesRepository(new ArrayList<>());
+        FormMapViewModel viewModel = new FormMapViewModel(FormMapViewModelTest.testForm1, testInstancesRepository);
+        activity.viewModelFactory = new TestFactory(viewModel);
+        activity.map = new TestMapFragment();
+
+        controller.setup().visible().start().resume();
```

Comment:
`setup` runs all the following method calls here (unless you're meaning to go through the lifecycle twice). I wrote the method 😉.

Reply:
Yikes, thank you! Did I get confused with `create`? I guess this is exactly why `ActivityScenario` is being pushed. Too may newbs like me putting together "creative" lifecycle calls.